src/service_props.py

```diff
@@ -0,0 +1,103 @@
+from dataclasses import dataclass
+from typing import List, Optional, Sequence
+
+from aws_cdk import aws_ecs as ecs
+
+CONTAINER_LOCATION_PATH_ID = "path://"
+
+
+@dataclass
+class ServiceSecret:
+    """
+    Holds onto configuration for the secrets to be used in the container.
+
+    Attributes:
+      secret_name: The name of the secret as stored in the AWS Secrets Manager.
+      environment_key: The name of the environment variable to be set within the container.
+    """
+
+    secret_name: str
+    """The name of the secret as stored in the AWS Secrets Manager."""
+
+    environment_key: str
+    """The name of the environment variable to be set within the container."""
+
+
+@dataclass
+class ContainerVolume:
+    """
+    Holds onto configuration for a volume used in the container.
+
+    Attributes:
+      path: The path on the container to mount the host volume at.
+      size: The size of the volume in GiB.
+      read_only: Container has read-only access to the volume, set to `false` for write access.
+    """
+
+    path: str
+    size: int = 15
+    read_only: bool = False
```

Comment:
One of the reasons why I added an `Attributes` section in both the docstring and under the attribute itself is what shows up in the "Mouse hover" windows in vscode.

When you mouse over an instance of the class it shows the class level docstring, but when you hover over an individual attribute it pulls from the docstring under the attribute.

Examples:
![image](https://github.com/user-attachments/assets/46f9e646-5908-431e-a688-d217a8df89e2)
![image](https://github.com/user-attachments/assets/d5d89e1b-4693-430d-b40c-9bd98970c222)

Reply:
done